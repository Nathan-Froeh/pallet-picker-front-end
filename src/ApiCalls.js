export const fetchAllProjects = () => {
  return fetch('http://localhost:3001/api/v1/projects')
    .then(res => res.json())
    .then(res => res)
    .catch (error => error.message)   
}

export const fetchPalettesById = (id) => {
  console.log(id)
  return fetch(`http://localhost:3001/api/v1/projects/${id}/palettes`)
    .then(res => res.json())
    .then(data => data)
    .catch(error => error.message)
}

