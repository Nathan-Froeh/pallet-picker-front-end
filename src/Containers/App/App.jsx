import React, {Component} from 'react';
import './App.scss';
import Generator from '../Generator/Generator';
import Projects from '../Projects/Projects'
import { 
  fetchAllProjects,
  fetchAllPalettes,
  deleteProject,
  deletePalette
 } from '../../ApiCalls';

export class App extends Component {
  constructor() {
    super()
    this.state = {
      projects: [],
      palettes: []
    }
  }
  componentDidMount () {
    this.getAllPalettes()
    this.getAllProjects()
  }

  getAllProjects = async () => {
    console.log(this.state.projects)
    await fetchAllProjects()
      .then(projects => this.setState({projects: projects}))
      .catch(error => error.message)
  }

  getAllPalettes = async () => {
    console.log(this.state.palettes)
    await fetchAllPalettes(this.state.project_id)
      .then(palettes => this.setState({palettes: palettes}))
      .catch(error => console.log(error.message))
  }

  deleteProject = (project_id) => {
    const remainingProjects = this.state.projects.filter(project => {
      return project.id !== project_id
    })
    deleteProject(project_id)
    this.setState({projects: remainingProjects})
  }

  deletePalette = (palette_id) => {
    const remainingPalettes = this.state.palettes.filter(palette => {
      return palette.id !== palette_id
    })
    deletePalette(palette_id)
    this.setState({palettes: remainingPalettes})
  }

  render () {
    return (
      <div className='App'>
        <h1>Happy Trees Palette Picker</h1>
        <Generator 
          projects={this.state.projects}
          getAllPalettes={this.getAllPalettes}
        />
        <Projects 
          projects={this.state.projects} 
          palettes={this.state.palettes}
          deleteProject={this.deleteProject}
          deletePalette={this.deletePalette}
          getAllProjects={this.getAllProjects}
        />
      </div>
    )
  }
}

export default App;
