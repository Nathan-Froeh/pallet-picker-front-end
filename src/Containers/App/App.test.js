
import React from 'react';
import {shallow} from 'enzyme';
import App from './App';
import {fetchAllProjects} from '../../ApiCalls';
import * as constants from '../../ApiCalls';


describe('App', () => {
  let wrapper, instance;

  beforeEach(() => {
    wrapper = shallow(<App/>)
    instance = wrapper.instance()
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('getAllProjects should call fetchAllProjects', async () => {
    constants.fetchAllProjects = jest.fn(() => Promise.resolve([]))
    expect(constants.fetchAllProjects).toHaveBeenCalledTimes(0)
    await instance.getAllProjects()
    expect( constants.fetchAllProjects).toHaveBeenCalledTimes(1)
  })

  it('getAllPalettes should call fetchAllPalettes', async () => {
    constants.fetchAllPalettes = jest.fn(() => Promise.resolve([]))
    expect(constants.fetchAllPalettes).toHaveBeenCalledTimes(0)
    await instance.getAllPalettes()
    expect( constants.fetchAllPalettes).toHaveBeenCalledTimes(1)
  })

  it('removeProject should call deleteProject', () => {
    constants.deleteProject = jest.fn(() => Promise.resolve([]))
    expect(constants.deleteProject).toHaveBeenCalledTimes(0)
    instance.removeProject()
    expect( constants.deleteProject).toHaveBeenCalledTimes(1)
  })

  it('removeProject should set projects state', () => {
    wrapper.setState({projects: [{id: 1, name: 'The barn'}]})
    expect(wrapper.state('projects')).toEqual([{id: 1, name: 'The barn'}])
    instance.removeProject(1)
    expect(wrapper.state('projects')).toEqual([])
  })

  it('removePalette should call deletePalette', () => {
    constants.deletePalette = jest.fn(() => Promise.resolve([]))
    expect(constants.deletePalette).toHaveBeenCalledTimes(0)
    instance.removePalette()
    expect( constants.deletePalette).toHaveBeenCalledTimes(1)
  })

  it('removePalette should set palette state', () => {
    wrapper.setState({palettes: [{id: 1, name: 'The barn'}]})
    expect(wrapper.state('palettes')).toEqual([{id: 1, name: 'The barn'}])
    instance.removePalette(1)
    expect(wrapper.state('palettes')).toEqual([])
  })

  

})