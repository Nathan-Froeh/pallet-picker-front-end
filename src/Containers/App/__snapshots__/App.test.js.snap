// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="App"
>
  <header>
    <h1>
      Palette Picker
    </h1>
  </header>
  <Generator
    getAllPalettes={[Function]}
    projects={Array []}
  />
  <Projects
    deletePalette={[Function]}
    deleteProject={[Function]}
    getAllProjects={[Function]}
    palettes={Array []}
    projects={Array []}
  />
</div>
`;
